webknossos/webknossos/skeleton/group.py

```diff
@@ -61,64 +79,87 @@ def add_graph(
 
     @property
     def children(self) -> Iterator[GroupOrGraph]:
+        """Returns all (immediate) children (groups and graphs) as an iterator."""
         return (child for child in self._children)
 
+    @property
+    def graphs(self) -> Iterator[GroupOrGraph]:
+        """Returns all (immediate) graph children as an iterator.
+        Use flattened_graphs if you need also need graphs within subgroups."""
+        return (child for child in self._children if isinstance(child, Graph))
+
+    @property
+    def groups(self) -> Iterator[GroupOrGraph]:
+        """Returns all (immediate) group children as an iterator.
+        Use flattened_groups if you need also need groups within subgroups."""
+        return (child for child in self._children if isinstance(child, Group))
+
     def add_group(
         self,
         name: str,
         _enforced_id: Optional[int] = None,
     ) -> "Group":
+        """Adds a (sub) group to the current group with the provided name."""
         new_group = Group(name, skeleton=self._skeleton, enforced_id=_enforced_id)
         self._children.add(new_group)
         return new_group
 
     def get_total_node_count(self) -> int:
+        """Returns the total number of nodes of all graphs within this group (and its subgroups)."""
         return sum(graph.number_of_nodes() for graph in self.flattened_graphs())
 
     def get_max_graph_id(self) -> int:
+        """Returns the highest graph id of all graphs within this group (and its subgroups)."""
         return max((graph.id for graph in self.flattened_graphs()), default=0)
 
     def get_max_node_id(self) -> int:
+        """Returns the highest node id of all nodes of all graphs within this group (and its subgroups)."""
         return max(
             (graph.get_max_node_id() for graph in self.flattened_graphs()),
             default=0,
         )
 
     def flattened_graphs(self) -> Generator[Graph, None, None]:
+        """Returns an iterator of all graphs within this group (and its subgroups)."""
         for child in self._children:
             if isinstance(child, Group):
                 yield from child.flattened_graphs()
             else:
                 yield child
 
     def flattened_groups(self) -> Generator["Group", None, None]:
+        """Returns an iterator of all groups within this group (and its subgroups)."""
         for child in self._children:
             if isinstance(child, Group):
                 yield child
                 yield from child.flattened_groups()
```

Comment:
Please refactor those to use `self.groups` in `flattened_groups` directly and `self.graphs` in `flattened_graphs`. E.g.
```suggestion
    def flattened_graphs(self) -> Iterator[Graph]:
        """Returns an iterator of all graphs within this group (and its subgroups)."""
        yield from self.graphs
        for group in self.groups:
            yield from group.flattened_graphs()

    def flattened_groups(self) -> Iterator["Group"]:
        """Returns an iterator of all groups within this group (and its subgroups)."""
        for group in self.groups:
            yield group
            yield from group.flattened_groups()
```